### ISSUE TITLE
Allow providing --tsserver-path

### ISSUE THREAD

**Issue by alecmev** [sublimelsp/LSP-typescript#23]
I'm using Yarn 2, which has no `node_modules` + TypeScript is incompatible with it out of the box, so Yarn puts a patched version in `project_root/.yarn/sdks`, and I'd like to have the language server use it.

Looking at `lsp_utils`, it appears there's no way to add any arguments:

https://github.com/sublimelsp/lsp_utils/blob/af3ee66115a231cc4b4d49610f9d176bc3edbad8/st3/lsp_utils/npm_client_handler_v2.py#L83
https://github.com/sublimelsp/lsp_utils/blob/af3ee66115a231cc4b4d49610f9d176bc3edbad8/st3/lsp_utils/npm_client_handler_v2.py#L117-L121

Ideas?

I suppose the proper fix would be to add Yarn-friendly resolution to the language server here:

https://github.com/theia-ide/typescript-language-server/blob/3114c557b58eb6b286e3c63f68ffe56c26800224/server/src/lsp-server.ts#L75-L94

But that repo is where PRs go to die... :smile:

**Comment by rchl** [comment]
Is there some argument that could be passed to the TS server that would help this case?

I think I agree that it's the server that should have support for that but I'm not sure how Yarn 2 works exactly.

**Comment by rchl** [comment]
> Is there some argument that could be passed to the TS server that would help this case?

Oh, I saw the subject now.

We could extend `lsp_utils` to read extra arguments from settings but I suppose that just passing plain path from settings won't work that well as you want to have different path per-project. Or eventually, a relative path that is resolved to absolute using project's root but that sounds awfully specific to TS server and I wouldn't want such code in a generic library.

**Comment by alecmev** [comment]
The path needs not be absolute, I'm using `--tsserver-path .yarn/sdks/typescript/lib/tsserver.js` without issues. It does assume that all projects are based on Yarn 2, but that's what per-project overrides are for.

**Comment by rchl** [comment]
Created https://github.com/sublimelsp/lsp_utils/pull/20 which should allow specifying `extra_binary_arguments` to provide extra arguments for the server binary.

Although I've just realized that it doesn't support project-specific settings so not sure if it's really worth it.

**Comment by alecmev** [comment]
Thank you! Well, it's good enough for me :smile: But yeah, would be nice to improve the server instead at some point.

**Comment by rchl** [comment]
The `command` can be fully overridden since https://github.com/sublimelsp/LSP-typescript/commit/4fc83b843532eef3dfa7e5b7de40b84419d3ada3 so I hope that will help here.

**Comment by alecmev** [comment]
Thank you! Works like a charm. For future readers, just put this in your `LSP-typescript`'s settings:

```json
"command": [
  "node",
  "${server_path}",
  "--stdio",
  "--tsserver-path",
  ".yarn/sdks/typescript/lib/tsserver.js"
]
```

Remember to `yarn add typescript` and `yarn dlx @yarnpkg/pnpify --sdk base` beforehand, as per [the docs](https://yarnpkg.com/advanced/editor-sdks).

**Comment by alecmev** [comment]
For the readers in the even further future, [this is no longer necessary as of a few weeks ago](https://github.com/theia-ide/typescript-language-server/pull/220)! :tada: Thanks, @tapayne88!

**Comment by rchl** [comment]
Just gonna clarify that that fix is not yet released as of `v0.5.4`. Should be in the next version.
